README.md

```diff
@@ -1,46 +1,123 @@
 
-common error of trying to generate a route with a variable without providing an argument for that route variabel value.
+* **Column 1** - This column gives the prefix for the route helper methods. In the current application ```posts``` and ```post``` are the prefixes for the methods that you can use throughout your applications. The two most popular method types are ```_path``` and ```_url```. So if we want to render a relative link path to our posts' index page the method would be ```posts_path``` or ```posts_url```. The difference between ```_path``` and ```_url``` is that ```_path``` gives the relative path and ```_url``` renders the full URL.
 
-passing data to dynamic routes from lowest level to highest level
+* **Column 2** - This is the HTTP verb
 
-using post_path(:id => @post.id)
+* **Column 3** - This column shows what the path for the route will be and what parameters need to be passed to the route. As you may notice the second row for the show route calls for an ID. When you pass the ```resources``` method the ```:show``` argument it will automatically create this route and assumes that you will need to pass the id into the URL string. Whenever you have id parameters listed in the path like this you will need to pass the route helper method an id, so our show route would look like this: ```post_path(@post)```. Notice how this is different than the index route of ```posts_path```.
 
-using post_path(@post.id)
+* **Column 4** - This column shows the controller and action with the syntax of ```controller#action```.
 
-using post_path(@post)
+One of the other nice things about utilize route helper methods is that they create predictable names for the methods. Once you get into day to day Rails development you will only need to run ```rake routes``` to find custom paths. Let's imagine that you takeover a legacy Rails application that was built with traditional routing conventions. If you see CRUD controllers for: newsletters, students, sales, offers, and coupons; you don't have to lookup the routes to know that you could call the index URLs for each resource below:
+
+* Newsletters - ```newsletters_path```
+
+* Students - ```students_path```
+
+* Sales - ```sales_path```
+
+* Offers - ```offers_path```
+
+* Coupons - ```coupons_path```
+
+
+## link_to Method
+
+All of our tests are currently passing, let's create a new Capybara spec. The scenario we will be building is to ensure that a link from the index page will point to that post's respective show page view template. The scenario is below:
+
+```ruby
+scenario 'index page links to post page' do
+  second = FactoryGirl.create(:second_post)
+  visit posts_path
+  expect(page).to have_link(second.title, href: post_path(second))
+end
+```
+
+This matcher will fail since our index page doesn't currently link to the show page. To fix this let's update the index page like so:
+
+```ERB
+<% @posts.each do |post| %>
+  <div><%= link_to post.title, "/posts/#{post.id}" %></div>
```

Comment:
Don't just drop them directly to the `link_to`. Show the normal `a href` way of doing things, then refactor to `link_to`. Also maybe showing them link_to in rails c to show what it outputs